TESTS/mbedtls/ecp/main.cpp

```diff
@@ -0,0 +1,299 @@
+
+    _TEST_ASSERT( mbedtls_mpi_read_string( &dA, 16, dA_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &xA, 16, xA_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &yA, 16, yA_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &dB, 16, dB_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &xB, 16, xB_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &yB, 16, yB_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &xZ, 16, xZ_str ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_read_string( &yZ, 16, yZ_str ) == 0 );
+
+    _TEST_ASSERT( mbedtls_ecp_mul( &grp, &R, &dA, &grp.G,
+                          &rnd_pseudo_rand, &rnd_info ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.X, &xA ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.Y, &yA ) == 0 );
+    _TEST_ASSERT( mbedtls_ecp_check_pubkey( &grp, &R ) == 0 );
+    _TEST_ASSERT( mbedtls_ecp_mul( &grp, &R, &dB, &R, NULL, NULL ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.X, &xZ ) == 0 );
+
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.Y, &yZ ) == 0 );
+    _TEST_ASSERT( mbedtls_ecp_check_pubkey( &grp, &R ) == 0 );
+
+    _TEST_ASSERT( mbedtls_ecp_mul( &grp, &R, &dB, &grp.G, NULL, NULL ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.X, &xB ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.Y, &yB ) == 0 );
+    _TEST_ASSERT( mbedtls_ecp_check_pubkey( &grp, &R ) == 0 );
+    _TEST_ASSERT( mbedtls_ecp_mul( &grp, &R, &dA, &R,
+                          &rnd_pseudo_rand, &rnd_info ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.X, &xZ ) == 0 );
+    _TEST_ASSERT( mbedtls_mpi_cmp_mpi( &R.Y, &yZ ) == 0 );
+    _TEST_ASSERT( mbedtls_ecp_check_pubkey( &grp, &R ) == 0 );
+
+exit:
+    mbedtls_ecp_group_free( &grp ); mbedtls_ecp_point_free( &R );
+    mbedtls_mpi_free( &dA ); mbedtls_mpi_free( &xA ); mbedtls_mpi_free( &yA ); mbedtls_mpi_free( &dB );
+    mbedtls_mpi_free( &xB ); mbedtls_mpi_free( &yB ); mbedtls_mpi_free( &xZ ); mbedtls_mpi_free( &yZ );
+    TEST_ASSERT(_assert == -1);
+}
+
+/* END_CASE */
+#define UTEST_FOR_ECP_TEST_VECTOR(id, dA_str, xA_str, yA_str, dB_str, xB_str, yB_str, xZ_str, yZ_str) \
+void utest_ ## id () {           \
+    ecp_test_vect(id, dA_str, xA_str, yA_str, dB_str, xB_str, yB_str, xZ_str, yZ_str); \
+} \
+Case case_ ## id (#id, utest_ ## id) \
+
+
+#if defined MBEDTLS_ECP_DP_SECP192R1_ENABLED
+UTEST_FOR_ECP_TEST_VECTOR(ecp_test_vect, 
+        MBEDTLS_ECP_DP_SECP192R1,
+        "323FA3169D8E9C6593F59476BC142000AB5BE0E249C43426",
+        "CD46489ECFD6C105E7B3D32566E2B122E249ABAADD870612",
+        "68887B4877DF51DD4DC3D6FD11F0A26F8FD3844317916E9A",
+        "631F95BB4A67632C9C476EEE9AB695AB240A0499307FCF62",
+        "519A121680E0045466BA21DF2EEE47F5973B500577EF13D5",
+        "FF613AB4D64CEE3A20875BDB10F953F6B30CA072C60AA57F",
+        "AD420182633F8526BFE954ACDA376F05E5FF4F837F54FEBE",
+        "4371545ED772A59741D0EDA32C671112B7FDDD51461FCF32");
```

Comment:
Note that this will not compile because the macro `UTEST_FOR_ECP_TEST_VECTOR` takes 9 commands and this is passing 10.